gateway-ha/src/test/java/io/trino/gateway/ha/router/TestRoutingGroupSelector.java

```diff
@@ -178,6 +178,21 @@ void testTrinoQueryPropertiesQueryType()
         assertThat(routingGroupSelector.findRoutingGroup(mockRequest)).isEqualTo("type-group");
     }
 
+    @Test
+    void testTrinoQueryPropertiesResourceGroupQueryType()
+            throws IOException
+    {
+        RoutingGroupSelector routingGroupSelector =
+                RoutingGroupSelector.byRoutingRulesEngine("src/test/resources/rules/routing_rules_trino_query_properties.yml", requestAnalyzerConfig);
+        String query = "CREATE TABLE cat.schem.foo (c1 int)";
+        Reader reader = new StringReader(query);
+        BufferedReader bufferedReader = new BufferedReader(reader);
+        HttpServletRequest mockRequest = prepareMockRequest();
+        when(mockRequest.getReader()).thenReturn(bufferedReader);
```

Comment:
We could inline some variables:
```java
        when(mockRequest.getReader())
                .thenReturn(new BufferedReader(new StringReader("CREATE TABLE cat.schem.foo (c1 int)")));
```

I would suggest adding a helper method in follow-up.